core/network/besu/contracts/interop/manageAssetAny.sol

```diff
@@ -140,17 +392,25 @@ contract AssetExchangeContract is ERC1155Holder {
     }
 
     // The receiver claims the ownership of an asset locked for them once they obtain the preimage of the hashlock
-    function claimAsset(bytes32 lockContractId, bytes32 preimage)
+    function claimAsset(bytes32 lockContractId, bytes memory claimInfoProtobuf)
         external
         returns (bool)
     {
         LockContract storage c = lockContracts[lockContractId];
-
+        AssetClaimHTLC memory params;
+        (, , params) = AssetClaimHTLCCodec.decode(
+            0,
+            claimInfoProtobuf,
+            uint64(claimInfoProtobuf.length)
+        );
+        bytes32 preimage = bytesTobytes32(
+            slice(params.hashPreimageBase64, 2, 32)
+        );
         // Check the validity of the claim
         require(c.status == LOCKED, "lockContract is not active");
         require(block.timestamp < c.expirationTime, "lockContract has expired");
         require(
-            c.hashLock == sha256(abi.encodePacked(preimage)),
+            c.hashLock == preimage,
```

Comment:
why this change? hashLock should not be equal to preimage, hashLock should be hash of preimage.

Also can you add if else conditions here to check the HashMechanism, and depending upon HashMechanism use sha256 only if HashMechanism is SHA256, else throw error?